src/vm-vcpu/src/vm.rs

```diff
@@ -318,7 +346,7 @@ impl<EH: 'static + ExitHandler + Send> KvmVm<EH> {
     // Configures the in kernel interrupt controller.
     // This function should be reused to configure the aarch64 interrupt controller (GIC).
     #[cfg(target_arch = "x86_64")]
-    fn setup_irq_controller(&self) -> Result<()> {
+    fn setup_irq_controller(&mut self) -> Result<()> {
```

Comment:
The mutability is already needed to initialize the `gic` field.

Reply:
It is only needed on aarch64, that's why I was asking.

Reply:
Got it. I changed both for symmetry, but have no problem with keeping the x86_64 version immutable.

Reply:
I've decided to keep `mut` here, because otherwise the clippy check would fail for x86 due to the unnecessary `mut` when defining the kvm object https://buildkite.com/rust-vmm/vmm-reference-ci/builds/1098#0181fd1d-c20d-44d2-8362-c7ee87481fa2/39-99